eth/tracers/native/call.go

```diff
@@ -176,6 +181,28 @@ func (t *callTracer) CaptureState(pc uint64, op vm.OpCode, gas, cost uint64, sco
 		}
 
 		data := scope.Memory.GetCopy(int64(mStart.Uint64()), int64(mSize.Uint64()))
+		if data == nil || uint64(len(data)) < mSize.Uint64() {
+			// memory not yet expanded
+			gasRequired, err := vm.MemoryGasCost(scope.Memory, mSize.Uint64())
+			if err != nil {
+				// overflow error. no logs will be emitted
+				return
+			}
+			if gas < gasRequired {
+				// memory expansion will fail: no logs will be emitted
+				return
+			}
```

Comment:
I think this is making the tracer too smart. Let's just put some hard upper boundary on the alloc we perform here. I think we do that already, at some similar place

Reply:
Okay.  I think the easiest behavior is just to assume the memory expansion will not fail and emit logs.